File: Navtrack.Core/ClientApp/src/apis/types/device/DeviceTypeModel.ts
import { ProtocolModel, DefaultProtocolModel } from "../protocol/ProtocolModel";

export type DeviceTypeModel = {
  id: number;
  manufacturer: string;
  model: string;
  displayName: string;
  protocol: ProtocolModel;
};

export const DefaultDeviceTypeModel : DeviceTypeModel = {
  id: 0,
  manufacturer: "",
  model: "",
  displayName: "",
  protocol: DefaultProtocolModel
};

File: backend/Navtrack.Api.Model/Devices/DeviceTypeModel.cs
using System.ComponentModel.DataAnnotations;
using Navtrack.Api.Model.Protocols;

namespace Navtrack.Api.Model.Devices;

public class DeviceTypeModel
{
    [Required]
    public string Id { get; set; }
    [Required]
    public string Manufacturer { get; set; }
    [Required]
    public string Model { get; set; }
    [Required]
    public string DisplayName => $"{Manufacturer} {Model}";
    [Required]
    public ProtocolModel Protocol { get; set; }
}